### PR TITLE
Add table `gcp_compute_ha_vpn_gateway` Closes #406

### LINKED ISSUE #406: Add table `gcp_compute_ha_vpn_gateway`

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Outputs:

project_id = "parker-aaa"
resource_aka = "gcp://compute.googleapis.com/projects/parker-aaa/regions/us-east1/vpnGateways/turbottest92884"
resource_name = "turbottest92884"
self_link = "https://www.googleapis.com/compute/v1/projects/parker-aaa/regions/us-east1/vpnGateways/turbottest92884"

Running SQL query: test-get-query.sql
[
  {
    "description": "Test VPN Gateway to validate integration test.",
    "kind": "compute#vpnGateway",
    "location": "us-east1",
    "name": "turbottest92884",
    "self_link": "https://www.googleapis.com/compute/v1/projects/parker-aaa/regions/us-east1/vpnGateways/turbottest92884"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "description": "Test VPN Gateway to validate integration test.",
    "kind": "compute#vpnGateway",
    "location": "us-east1",
    "name": "turbottest92884",
    "self_link": "https://www.googleapis.com/compute/v1/projects/parker-aaa/regions/us-east1/vpnGateways/turbottest92884"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "gcp://compute.googleapis.com/projects/parker-aaa/regions/us-east1/vpnGateways/turbottest92884"
    ],
    "title": "turbottest92884"
  }
]
✔ PASSED

POSTTEST: tests/gcp_compute_ha_vpn_gateway

TEARDOWN: tests/gcp_compute_ha_vpn_gateway

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```sql
> select
  name,
  id,
  description,
  location,
  self_link,
  kind
from
  gcp_compute_ha_vpn_gateway;
+-------+---------------------+-------------+-------------------------+-------------------------------------------------------------------------------------------------------------+--------------------+
| name  | id                  | description | location                | self_link                                                                                                   | kind               |
+-------+---------------------+-------------+-------------------------+-------------------------------------------------------------------------------------------------------------+--------------------+
| test1 | 6193525178265654635 |             | northamerica-northeast1 | https://www.googleapis.com/compute/v1/projects/parker-aaa/regions/northamerica-northeast1/vpnGateways/test1 | compute#vpnGateway |
+-------+---------------------+-------------+-------------------------+-------------------------------------------------------------------------------------------------------------+--------------------+
> 
> select
  name as vpn_gateway_name,
  i ->> 'id' as vpn_interface_id,
  i ->> 'ipAddress' as vpn_interface_ip_address
from
  gcp_compute_ha_vpn_gateway g,
  jsonb_array_elements(vpn_interfaces) i;
+------------------+------------------+--------------------------+
| vpn_gateway_name | vpn_interface_id | vpn_interface_ip_address |
+------------------+------------------+--------------------------+
| test1            | <null>           | 34.124.13.35             |
| test1            | 1                | 34.104.77.210            |
+------------------+------------------+--------------------------+
```
</details>

### CODE REVIEW

File: docs/tables/gcp_compute_ha_vpn_gateway.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Table: gcp_compute_ha_vpn_gateway
+
+Represents a VPN gateway running in GCP. This virtual device is managed by Google, but used only by you. This type of VPN Gateway allows for the creation of VPN solutions with higher availability than classic Target VPN Gateways.
+
+### Basic info
```
Comment (misraved) [inline docs/tables/gcp_compute_ha_vpn_gateway.md:5]:
```suggestion
## Examples

### Basic info
```

File: gcp/plugin.go [hunk 2]
```diff
@@ -69,6 +69,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"gcp_compute_target_pool":                                 tableGcpComputeTargetPool(ctx),
 			"gcp_compute_target_ssl_proxy":                            tableGcpComputeTargetSslProxy(ctx),
 			"gcp_compute_target_vpn_gateway":                          tableGcpComputeTargetVpnGateway(ctx),
+			"gcp_compute_ha_vpn_gateway":                              tableGcpComputeHaVpnGateway(ctx),
```
Comment (misraved) [inline gcp/plugin.go:72]: Could you please sort the table list?

File: gcp/table_gcp_compute_ha_vpn_gateway.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+	return &plugin.Table{
+		Name:        "gcp_compute_ha_vpn_gateway",
+		Description: "GCP Compute VPN Gateway",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("name"),
+			Hydrate:    getComputeHaVpnGateway,
+		},
+		List: &plugin.ListConfig{
+			Hydrate:           listComputeHaVpnGateways,
+			ShouldIgnoreError: isIgnorableError([]string{"403"}),
```
Comment (misraved) [inline gcp/table_gcp_compute_ha_vpn_gateway.go:26]: Should we ignore the `403` error code?

**Review by misraved — COMMENTED** [review]
Please make changes wherever applicable 👍. Thanks!!

File: gcp/table_gcp_compute_ha_vpn_gateway.go [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "A friendly name that identifies the resource.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The unique identifier for the resource.",
+				Type:        proto.ColumnType_INT,
```
Comment (ParthaI) [inline gcp/table_gcp_compute_ha_vpn_gateway.go:37]: Should it be an integer type?

File: gcp/table_gcp_compute_ha_vpn_gateway.go [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+				Description: "Labels for this resource.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "vpn_interfaces",
+				Description: "The list of VPN interfaces associated with this VPN gateway.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// standard steampipe columns
```
Comment (ParthaI) [inline gcp/table_gcp_compute_ha_vpn_gateway.go:99]:
```suggestion
			//  Steampipe standard columns
```

File: gcp/table_gcp_compute_ha_vpn_gateway.go [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getVpnGatewayAka,
+				Transform:   transform.FromValue(),
+			},
+
+			// standard gcp columns
```
Comment (ParthaI) [inline gcp/table_gcp_compute_ha_vpn_gateway.go:120]:
```suggestion
			// GCP standard columns
```

File: gcp/table_gcp_compute_ha_vpn_gateway.go [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+	// Get project details
+	getProjectCached := plugin.HydrateFunc(getProject).WithCache()
+	projectId, err := getProjectCached(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	project := projectId.(string)
+
+	var vpnGateway compute.VpnGateway
+	name := d.KeyColumnQuals["name"].GetStringValue()
```
Comment (ParthaI) [inline gcp/table_gcp_compute_ha_vpn_gateway.go:200]: Please add an empty check for it.

**Review by ParthaI — APPROVED** [review]
LGTM

**Pull request merged**